Nginx revproxy 4 nexus

q-uest opened the pull request:
Note:
Besides getting the certificate signed by a local CA and adding it to the trusted list of certificates on the client OS, still getting the Warning in (chrome) browser.

ranganathan-ramanathan committed:
Ansible Playbook for Nginx Reverse proxy for Nexus

ranganathan-ramanathan committed:
Ansible playbook for Nginx Reverse Proxy for Nexus

projectdec2021 commented on revprxy.yaml:
```diff
@@ -0,0 +1,160 @@
+- hosts: 34.121.46.70
```
this host ip should goto inventory file

q-uest replied:
Replaced IP with Group name

projectdec2021 commented on revprxy.yaml:
```diff
@@ -0,0 +1,160 @@
+- hosts: 34.121.46.70
+  gather_facts: no
+  remote_user: ubuntu
```
any specific reason we are going for ubuntu user?

q-uest replied:
sure, it does not look appropriate having this @ ubuntu.....creating a specific user named, "nginx"  and running the playbook there instead.

projectdec2021 commented on revprxy.yaml:
```diff
@@ -0,0 +1,160 @@
+    - name: ensure nginx is at the latest version
+      apt:
+        name=nginx
+        state=latest
```
Lets got with either = format or : format while sending module parameters

projectdec2021 commented on revprxy.yaml:
```diff
@@ -0,0 +1,160 @@
+      openssl_csr:
+        path: /root/example_com.csr
+        privatekey_path: "{{ example_com_key.filename }}"
+        common_name: 34.121.46.70
```
This should not be hardcoded, should be used as variable

q-uest replied:
Used variables as suggested.

projectdec2021 commented on revprxy.yaml:
```diff
@@ -0,0 +1,160 @@
+        create: yes
+        block: |
+          ssl_protocols TLSv1.3;
+          ssl_prefer_server_ciphers on;
```
keep this as external file and use template or copy module for this

q-uest replied:
Done.

projectdec2021 commented on revprxy.yaml:
```diff
@@ -0,0 +1,160 @@
+
+    - name: ensure file exists
+      file:
+        path: /etc/nginx/sites-available/nexus
```
Its redundant, can be removed

q-uest replied:
removed in the new commit

projectdec2021 commented on revprxy.yaml:
```diff
@@ -0,0 +1,160 @@
+        path: "/etc/nginx/sites-available/nexus"
+        block: |
+          server {
+            listen 443 ssl;
```
keep this as external file and use template or copy module for this

projectdec2021 commented on revprxy.yaml:
```diff
@@ -0,0 +1,160 @@
+        state: absent
+
+    - name: Remove file nginx default config file
+      ansible.builtin.file:
```
module name should be only file

projectdec2021 commented on revprxy.yaml:
```diff
@@ -0,0 +1,160 @@
+
+    - name: Create a symbolic link
+      ansible.builtin.file:
+        src: /etc/nginx/sites-available/nexus
```
module name should be only file

q-uest replied:
updated in the new commit

projectdec2021 commented on revprxy.yaml:
```diff
@@ -0,0 +1,160 @@
+  handlers:
+    - name: Restart nginx service
+      ansible.builtin.service:
+        name: nginx
```
module name should be only service

q-uest replied:
updated

q-uest committed:
Update revprxy.yaml

updated the play book per the PR review

q-uest committed:
Update revprxy.yaml

Removed the redundant task of "touch"-ing the config file, named 'nexus' per the PR review

ranganathan-ramanathan committed:
updated per PR review

q-uest reviewed (commented):
1 more new playbook has been added for creating user-nginx, updated the revprxy.yaml and added templates required for configuration files etc.

q-uest committed:
Create my_vault.yml

ranganathan-ramanathan committed:
updated playbooks & added logs

ranganathan-ramanathan committed:
updated cre_nginx.yaml to pass correct value to pubkey_file variable

ranganathan-ramanathan committed:
Combined the templates, now there is only one template file

ranganathan-ramanathan committed:
adding log files of the latest run
